lib.rs

```diff
@@ -258,25 +258,25 @@ impl<A: Array> SmallVecData<A> {
     #[inline]
     unsafe fn inline(&self) -> &A {
         match *self {
-            SmallVecData::Inline(ref a) => a,
+            SmallVecData::Inline(ref a) => &*a.as_ptr(),
```

Comment:
I believe it's unsound to return an `&A` that points to a possibly-uninitialized `A`.  This method should be changed to return a raw pointer. (The callers of this method only need raw pointers anyways.)